### DISCUSSION TITLE
Developer portal proposal and community plans

### DISCUSSION THREAD

**Discussion by deboragaleano** [dhis2/notes#213] [Ideas]
1. This is a [proposal to revamp](https://docs.google.com/document/d/1kc-AeYeggMWbnevW5Z0ZfjQJKIFNJaS3Qr1o2l5GMJQ/edit?usp=sharing) the existing Developer Portal and work towards the following goals:

- Consolidate documentation
- Improve the existing docsite tooling (I'm suggesting that we use [Docusaurus](https://v2.docusaurus.io/)) 
- Make documentation easy to update and maintain
- Improve the design and website structure

Here's a [live demo](https://deboragaleano.github.io/dev-portal-demo/) using Docusaurus for the new Developer Portal (still needs improvement!)

2. Community Outreach 

This is a draft of the [community plan](https://docs.google.com/document/d/1uCxmZJ9NpluQ7ymMoCftX9m7554kTTGWsdo12UO9J_8/edit?usp=sharing) and activities to carry out so as to promote the DHIS2 Developer Community. Some suggestions: 

- Create a dedicated, developer-focused space for support and discussion (Slack channel)
- Organize informal meetups for and by DHIS2 developers 

I'd love to get your feedback on any of these topics so please feel free! :)

**Comment by cooper-joe** [comment]
Thanks for the proposal @deboragaleano. Happy to see the documentation being consolidated. I have some comments about the design section of the proposal. From the proposal:

> The plan is to integrate the new DHIS2 website’s design in terms of color palette to maintain a consistent look and feel across DHIS2 websites.

I agree that it is important that the developer portal has a level of consistency with the DHIS2 branding, however I do not think the design of [DHIS2.org](https://dhis2.org/) should be so closely replicated on the Developer Portal.

The Developer Portal and [DHIS2.org](https://dhis2.org/) differ in at least two meaningful ways:
- **target audience**: DHIS2.org is for learning about DHIS2 and caters to all visitors that want to learn more about DHIS2; the target audience is vast. The Developer Portal has a more limited audience: developers working with DHIS2.
- **main reasons/use-cases for visiting**: DHIS2.org covers many different use cases and content types, the Developer Portal is more focused in it's purpose: serving useful information, guides and documentation for DHIS2 developers.

These differences in intention and audience should be reflected in differences in design too. Practically speaking, this covers:
- navigation (not using the same style top navigation bar), 
- font size
- color palette
- communication style
- illustrations/icons/visual style 

Consider the (soon to be released) updated [DHIS2 documentation](https://docs.dhis2.org/en/home.html). The design of the documentation intentionally differs from [DHIS2.org](https://dhis2.org/):
- the navigation bar is darker, a clear and instant visual clue that "This is not DHIS2.org, if you are looking for that". This helps visitors quickly orient themselves and understand if they are where they intend to be.
- the navigation bar offers some simple choices, and is therefore smaller and lower impact. The DHIS2.org navigation bar is larger because it needs to handle full-width dropdowns that cover many topics.
- the typography is tighter (smaller, reduced line-height, reduced headings) to allow for the consumption of more content. The documentation portal offers much more long-form content that needs to be browsed and scanned quickly. DHIS2.org is more focused on summary-style content blocks. It makes sense to apply different typographic treatments.
- color/illustration/visual style is much more subdued and serious. The documentation is most important, not any kind of visual brand positioning. DHIS2.org must concern itself more with brand presentation, the documentation portal does not need to do this as much.

My point being, I think the visual design of the Developer Portal should be considered in itself, not just inherit the [DHIS2.org](https://dhis2.org/) design as [DHIS2.org](https://dhis2.org/)-connected domain. I think the Developer Portal is important enough that it deserves design attention to make sure the design reflects the goals.

As a starting point, I would suggest a design that is more consistent with the [Documentation](https://docs.dhis2.org/en/home.html), rather than [DHIS2.org](https://dhis2.org/). More design work would mean not meeting the schedule outlined in the proposal, but I think it would be worth it.

**Reply by deboragaleano** [reply]
Thanks so much @cooper-joe for your feedback and suggestions! This is great and I agree with you that the design of the Developer Portal/Hub should be considered in itself.

Since I'm preparing a live demo for the webinar next Monday, I think I'd focus on the structure and the content of the portal more than the visual style for now (though I'd make a few changes that's more consistent with the Documentation as suggested) and then it would be great if we could discuss this further if you're available in the coming days :)

**Reply by cooper-joe** [reply]
Sounds good to me @deboragaleano - let me know when would be a good time to discuss. I'm happy to contribute to the design by modifying the template css.
